app/services/IdentityMatchService.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import config.AppConfig
+import connectors.IdentityMatchConnector
+import exceptions.{InvalidIdMatchRequest, InvalidIdMatchResponse, LimitException}
+import javax.inject.Inject
+import models.{IdMatchError, IdMatchRequest, IdMatchResponse}
+import models.api1585.IdMatchApiResponseSuccess
+import repositories.IndividualCheckRepository
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class IdentityMatchService @Inject()(val connector: IdentityMatchConnector,
+                                     val repository: IndividualCheckRepository,
+                                     val appConfig: AppConfig) {
+
+
+  def matchId(request: IdMatchRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[IdMatchError, IdMatchResponse]] = {
+
+    limitedMatch(request).recoverWith {
+      case ex: LimitException => getErrorResponse(ex.getLocalizedMessage)
+      case _: InvalidIdMatchResponse => getErrorResponse("Something went wrong")
```

Comment:
Ah - you have replied with the generic error 👍

Reply:
Yeah, I thought if I handled them explicitly it would be easier to expand the code in future 🙂